src/vm/prestub.cpp

```diff
@@ -443,7 +443,11 @@ PCODE MethodDesc::MakeJitWorker(COR_ILMETHOD_DECODER* ILHeader, DWORD flags, DWO
                 }
                 else
```

Comment:
You will incorrectly go into this code when `&& (! fBackgroundThread)` is false. This check needs to be moved up.